src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -280,37 +280,167 @@ def parse_known_args(self, args=None, namespace=None):
         self._namespace, self._raw_arguments = super().parse_known_args(args=args, namespace=namespace)
         return self._namespace, self._raw_arguments
 
-    def _check_value(self, action, value):
+    def _get_extension_command_tree(self):
+        from azure.cli.core._session import EXT_CMD_TREE
+        import os
+        VALID_SECOND = 3600 * 24 * 10
+        # self.cli_ctx is None when self.prog is beyond 'az', such as 'az iot'.
+        # use cli_ctx from cli_help which is not lost.
+        cli_ctx = self.cli_ctx or (self.cli_help.cli_ctx if self.cli_help else None)
+        if not cli_ctx:
+            return None
+        EXT_CMD_TREE.load(os.path.join(cli_ctx.config.config_dir, 'extensionCommandTree.json'), VALID_SECOND)
+        if not EXT_CMD_TREE.data:
+            import requests
+            from azure.cli.core.util import should_disable_connection_verify
+            try:
+                response = requests.get(
+                    'https://azurecliextensionsync.blob.core.windows.net/cmd-index/extensionCommandTree.json',
+                    verify=(not should_disable_connection_verify()),
+                    timeout=300)
+            except Exception as ex:  # pylint: disable=broad-except
+                logger.info("Request failed for extension command tree: %s", str(ex))
```

Comment:
Would log at warning be better?

Reply:
I will modify in upcoming PR. For now, let's just fallback to previous behavior when something is wrong in this feature.